File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/c++/over.cc
#include "over.h"
#include "Programa.h"
#include <stack>
using namespace std;

void Over::ejecutar(Programa* programa){
    stack<int>& pila = programa->getPila();
  
    int c = pila.top();
    pila.pop();
    int h = pila.top();
    pila.pop();

    pila.push(h);
    pila.push(c);
    pila.push(h);

    programa->inc();
}

string Over::mostrar(){
    return "over";
}

File: OOP-and-Functional-Programming_Samples/OOP/Simulated_Assembly_instructions_and_Samples/java/Over.java
public class Over extends Instruccion{

    public void ejecutar(Programa programa){

        int c = programa.getPila().pop();
        int h = programa.getPila().pop();
        programa.getPila().push(h);
        programa.getPila().push(c);
        programa.getPila().push(h);

        programa.inc();
    }

    public String mostrar(){
        return "over";
    }
}